server/models.py

```diff
@@ -0,0 +1,94 @@
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy.orm import validates, relationship
+from sqlalchemy import Column, Integer, String, ForeignKey
+import re
+
+db = SQLAlchemy()
+
+class User(db.Model):
+    __tablename__ = 'users'
+
+    user_id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(50), nullable=False, unique=True)
+    email = db.Column(db.String(120), nullable=False, unique=True)
+    password_hash = db.Column(db.String(128), nullable=False)
+    role = db.Column(db.String(10))
+    is_active = db.Column(db.Boolean(1), default=True)
+    confirmed_admin = db.Column(db.Boolean(1), default=False) 
+    
+    def __repr__(self):
+        return f'<User {self.username}>'
+    
+class Invitations(db.Model):
+    __tablename__ = 'invitations'
+
+    invitation_id = db.Column(db.Integer, primary_key=True)
+    token = db.Column(db.String(150), nullable=False, unique=True)
+    email = db.Column(db.String(120), nullable=False, unique=True)
+    created_at = db.Column(db.DateTime, server_default=db.func.now())
+    expiry_at = db.Column(db.DateTime, onupdate=db.func.now())
+    is_used = db.Column(db.Boolean(1), default=True)
+    
+    def __repr__(self):
+        return f'<Invitation {self.invitation_id}>'
+
+class Store(db.Model):
+    __tablename__ = 'stores'
+
+    store_id = db.Column(db.Integer, primary_key = True)
+    store_name = db.Column(db.String(50), nullable = False)
+    location = db.Column(db.String(50), nullable = False)
+    
+    # Relationship to access products associated with a store
+    store = relationship('Store', backref='products')
+
+
+def __repr__(self):
+        return f'Store(id={self.store_id}, name={self.store_name}, name={self.location})'
+
```

Comment:
This function should be part of the class, hence indent it accordingly.

Reply:
Correciton made to the indentation